---
title: "Manuscript Draft: Aim3 Leaf Traits"
date: today
bibliography: VBL_library.bib
csl: functional-ecology.csl
author:
  - name: Bolívar Aponte Rolón
    orcid: 0000-0002-2544-4551
    email: baponterolo@tulane.edu
    corresponding: true
    affiliation: 
      - id: tu
        name: Tulane University
        department: Ecology and Evolutionary Biology
        city: New Orleans
        state: LA
        country: US
        url: www.tulane.edu
  - name: Mareli Sánchez Juliá
    #orcid: 
    email: msanchezjulia@tulane.edu
    corresponding: false
    affiliation: 
      - id: tu
        name: Tulane University
        department: Ecology and Evolutionary Biology
        city: New Orleans
        state: LA
        country: US
        url: www.tulane.edu
  - name: A. Elizabeth Arnold
    orcid: 0000-0002-7013-4026
    email: arnold@ag.arizona.edu
    corresponding: false
    affiliation: 
      - id: ua
        name: University of Arizona
        department: School of Plant Sciences
        city: Tucson
        state: AZ
        country: US
        url: www.arizona.edu
      - id: ua2
        name: University of Arizona
        department: Department of Ecology and Evolutionary Biology
        city: Tucson
        state: AZ
        country: US
        url: www.arizona.edu  
  - name: Sunshine A. Van Bael
    orcid: 0000-0001-7317-3533
    email: svanbael@tulane.edu
    corresponding: false
    affiliation: 
      - ref: tu
format: 
  pdf:
    documentclass: scrartcl
    include-in-header:
        text: '\usepackage{lineno}\linenumbers'
    linestretch: 2.0
    margin-left: 1in
    margin-right: 1in
    margin-top: 1in
    margin-bottom: 1in
    mainfont: TeX Gyre Termes #Alternative to Times New Roman
    sansfont: TeX Gyre Termes #Another option is Liberation Serif
    number-sections: true
    citation-package: biblatex
editor: source
execute:
  echo: false
warning: false
---

```{r}
#| echo: false
#| eval: true
knitr::opts_chunk$set(ft.tabcolsep=0, ft.latex.float = "none")
```

# Abstract

# Keywords:

endophytes, atta colombica, herbivory, calonectria, pathogen, foliar, fungal, symbionts

# Introduction

# 

# Materials and Methods

## Field

Growth and host plant inoculation seven tropical tree species was conducted at the greenhouses in the Gamboa Research Station, Smithsonian Tropical Research institute, Republic of Panama. The species, *Theobroma cacao*, *Dypterix* sp., *Lacmellea panamensis*, *Apeiba membranacea*, *Heisteria concinna*, *Chrysophyllum caimito*, and *Cordia alliodora* were chosen due to their variance in leaf traits (J.Wright unpublished data) and the availability of seeds in January- April 2019. Seeds of tree species were collected from the forest floor and grown in the greenhouse. Seedlings were kept in a chamber made out of PVC and clear plastic to prevent inoculation from spore fall inside the greenhouse. NEEDS INFORMATION ON THE SOIL MIXTURE AND AUTOCLAVING PROTOCOL. Seedlings reached a minimum of 4 true leaves before endophyte inoculation. Then 10 individual plants of each species were exposed to 10 nights of spore fall to achieve a high endophyte load (E+) and 10 homologous plants were kept inside the greenhouse plastic chamber to maintain a low endophyte load (*E-*) (Fig. ? MAKE A DIRAGRAM?). Plants exposed to spore fall were placed near (\~10 m) the forest edge at dusk (\~18:OO hours) and returned to the greenhouse at dawn (\~07:00 hours) [@bittleston2011].

### Leaf trait measurements

Three mature leaves were haphazardly collected from each of the individual plants in each treatment (E+, E-) within 7-10 days after inoculation treatment. Anthocyanin (ACI) content and leaf thickness (LT) were measured while the leaf was still attached to the plant. We measured anthocyanin content with ACM-200plus (Opti-Sciences Inc. Hudson, New Hampshire, U.S.A.) on three haphazardly selected locations (working from the petiole out to the leaf tip) on the leaf surface of three haphazardly selected leaves for a total of nine measurements per plant [@tellezTraits2022]. The ACM-200 calculates an anthocyanin content index (ACI) value from the ratio of % transmittance at 931 nm/% transmittance at 525 nm [@tellezRedCol2016, @opti-sciencesinc.] . On compound leaves (i.e., *Dypterix* sp.) we measured at three different leaflets. Leaf thickness (μm) was measured with a Mitutoyo 7327 Micrometer Gauge (Mitutoyo, Takatsu-ku, Kawasaki, Japan) in sthe same manner as the anthocyanin measurements, taking care to avoid major and secondary veins. After anthocyanin and leaf thickness measurements were completed, we removed the leaves from their stems, placed them inside a plastic bag (i.e. ZiplocⓇ), place in an ice chest and moved them to the lab for further measurements. Leaf punch strength (LPS) was measured with an Imada DST-11a digital force gauge (Imada Inc., Northbrook, IL, United States) by conducting punch-and-die tests with a sharp-edged cylindrical steel punch (2.0 mm diameter) and a steel die with a sharp-edged aperture of small clearance (0.05 mm). The leaf punch measurements were taken by puncturing the leaf lamina at the base, mid-leaf and tip on both sides of the mid-vein, avoiding minor leaf veins when possible [@tellezTraits2022]. Once leaf toughness was measured, we used a 7 mm diameter punch hole to puncture disks for leaf mass per area (LMA) measurements. We collected one three disks per leaf (see Supplementary material for details). The disk punches dried at 60 °C for 48-72 hours. before being weighed.

### Leaf tissue preparation for molecular work

The selected leaves were also used to profile endophyte community composition, abundance, and richness via amplicon sequencing (Illumina MiSeq). The leaf tissue remaining after the leaf trait measurements had the main vein and margins excised so that only the lamina remained. The lamina was haphazardly cut into 2 x 2 mm segments, enough to obtain a total of 16, and surface sterilized by sequential rinsing in 95% ethanol (10 s), 0.5 NaOCl (2 mins) and 70% ethanol (2 mins), as per [@arnold2003; @higgins2014; @tellezTraits2022]. After, leaves were air-dried briefly under sterile conditions. Sixteen leaf segments per leaf, a total of forty-eight leaf segments per plant, were plated in 2% malt extract agar (MEA), sealed with Parafilm M (Bemis Company Inc., U.S.A.) and incubated at room temperature. The cultured leaf segments were used to estimate endophyte colonization of *E-* and *E+* leaves. The presence or absence of endophytic fungi in the leaf cuttings was assessed 7 days after plating. The remaining sterilized leaf lamina was preserved in sterile 15 mL tubes with \~ 10 mL CTAB solution (1 M Tris--HCl pH 8, 5 M NaCl, 0.5 M EDTA, and 20 g CTAB). Leaf tissue in CTAB solution was used for amplicon sequencing (described in detail below). All leaf tissue handling was performed in a biosafety cabinet with all surfaces sterilized by exposure to UV light for 30 minutes and cleaned sequentially in between samples with 95% ethanol, 0.5% NaOCl and 70% ethanol to prevent cross contamination.

## Amplicon sequencing

Leaf tissue in CTAB solution was stored for 2 months at room temperature prior to being placed at -80 C for 3 months before extracting DNA. In preparation for DNA extraction, we decontaminated all instruments, materials, and surfaces with DNAway (Molecular BioProducts Inc., San Diego, CA, United States), 95% Ethanol, 0.5 % NaOCl, and 70 % Ethanol, and subsequently treated with UV light for 30 minutes in biosafety cabinet. We then transferred 0.2 -- 0.3 g of leaf tissue into duplicate sterile 2mL tubes, resulting in 2 subsamples. Total genomic DNA from subsamples was extracted as described in U'Ren & Arnold [-@uren2017]. In brief, we added two sterile 3.2 mm stainless steel beads to each tube and proceeded to lyophilize samples for 72 hours to fully remove CTAB content from tissue. After this period, we submerged the sample tubes in liquid nitrogen for 30s and proceeded to homogenize samples to a fine powder for 45 s in FastPrep-24 Tissue and Cell Homogenizer (MP Biomedicals, Solon, OH, USA). Afterwards, we repeated the decontamination procedure described before and used QIAGEN DNeasy 96 PowerPlant Pro-HTP Kit [@uren2017] (QIAGEN, Valencia, CA, USA). After all genomic DNA was extracted, we pooled the subsamples for each individual sample before amplification. We used sterile equipment and pipettes with aerosol-resistant tips with filters in all steps before amplification. We followed a two-step amplification approach previously described by Sarmiento et al. [-@sarmiento2017] and U´Ren & Arnold [-@uren2017]. We used primers for the fungal ITSrDNA region, ITS1f (5'-CTTGGTCATTTAGAGGAAGTAA-3') and ITS4 (5'-TCCTCCGCTTATTGATATGC-3') with modified universal consensus sequences CS1 and CS2 and 0--5 bp for phase-shifting. Every sample was amplified in two parallel reactions containing 1-2 µL of DNA template [@uren2017; see also @tellezTraits2022]. We visualized PCR (PCR1) reactions with SYBR Green 1 (Invitrogen, Carlsbad, CA, USA) on 2% agarose gel [@oita2021]. Based on the electrophoresis band intensity, we combined parallel PCR1 reactions and diluted 5 µL of amplicon product with molecular grade water to standardize to a concentration of 1:15 [@tellezTraits2022; @sarmiento2017 for details]. We included DNA extraction blanks and PCR1 negatives in this step. We used a separate set of sterile pipettes, tips, and equipment to reduce contamination. We used a designated PCR area to restrict contact with pre-PCR materials [@oita2021].

We used 1 µL of PCR1 product from samples and negative control for a second PCR (PCR2) with barcode adapters (IBEST Genomics Resource Core, Moscow, ID, USA). Each PCR2 reaction (total 15 µL) contained 1X Phusion Flash High Fidelity PCR Master Mix, 0.075 µM of barcoded primers (forward and reverse pooled at a concentration of 2 µM) and 0.24mg/mL of BSA following Sarmiento [-@sarmiento2017] and U'Ren & Arnold [-@uren2017]. Before final pooling for sequencing, we purified the amplicons using Agencourt AMPure XP Beads (Beckman Coulter Inc, Brea, CA USA) to a ratio of 1:1 following the manufacturer's instructions. The products were evaluated with Bio Analyzer 2100 (Agilent Technologies, Santa Clara, CA, USA) [@tellezTraits2022]. We quantified the samples through University of Arizona Genetics Core, and subsequently diluted them to the same concentration to prevent over representation of samples with higher concentration, see (CITATION). Amplicons were normalized to 1 ng/µL, then pooled 2 µL of each for sequencing. No contamination was detected visually or by fluorometric analysis. To provide robust controls we combined 5 µL of each PCR1 negative and the DNA extraction blanks and sequenced them as samples. Ultimately, we combined samples into a single tube with 20 ng/µL of amplified DNA with barcoded adapters for sequencing on the Illumina MiSeq platform with Reagent Kit v3 (2 × 300 bp) following protocols from the IBEST Genomics Resource Core at the University of Idaho, USA. Again, we included the DNA extraction blanks and two PCR1 negatives and sequenced with samples. Sequencing yielded 3,778,081 total ITS1 reads.

### Replication Statement (maybe goes before stattistical analyses)

*In the Materials and Methods section (before the description of the data analyses), authors **MUST** state i) the scale at which they seek to make inferences (for example at the level of species, populations or experimental units); ii) the scale(s) at which their treatment or factor of interest is(are) applied; and iii) the number of replicates for each level of treatment or factor. We have provided a table template for including this information, which the editors will use to decide whether the authors' inferences are supported and should therefore be peer-reviewed. Manuscripts without this table will be returned to authors.*

### Mock Communities

We processed and sequenced 12 mock communities following the methods described above. This allowed us to assess the quality of our NGS data set. We used two mock communities that consisted of PCR product from DNA extractions of 32 phylogenetically distinct fungi, representing lineages that are typically observed as endophytes: Ascomycota, Basidiomycota, Zygomycota and Chytridiomycota [@oita2021; see @daru2019 for details]. In brief, we used six mock communities with equimolar concentrations of DNA from all 32 fungal taxa and another six mock communities with tiered concentrations of DNA from the same fungal taxa [@daru2019]. Each mock community was sequenced five times (i.e., five replicates) [@oita2021]. The read abundance from the equimolar and tiered communities was positively associated with the expected read number (with replicates as a random factor: R2Adj = 0.87, P = XXXX, see Supplementary Material). Using mock communities allowed us to evaluate the sequencing effectiveness in communities with known composition and structure [@bowman2021]. Henceforth, we used read abundance as a relevant proxy for biological OTU abundance [@uren2019].

### Bioinformatic analyses

We used VSEARCH (v2.14.1) for *de novo* chimera detection, dereplication and sequence alignment. VSEARCH is an open-source alternative to USEARCH that uses an optimal global aligner (full dynamic programming Needleman-Wunsch), resulting in more accurate alignments and sensitivity [@rognes2016]. For mock communities and experimental samples, we used forward reads (ITS1) for downstream bioinformatics analyses due to their high quality, rather than reverse reads (ITS4). Following Sarmiento et al. [-@sarmiento2017], we concatenated all reads in a single file and used FastQC reports to assess Phred scores above 30 and determine the adequate length of truncation. We processed 892,713 of sequence reads from mock communities and 3,778,081 from experimental samples. We truncated mock community and experimental sample reads to a length of 250 bp with command `fast_trunclen` and filtered them at a maximum expected error of 1.0 with command `fast_maxee`. We then clustered unique sequence zero radius OTUs (that is, zOTUs; analogous to amplicon sequence variants [@callahan2016]), by using commands `derep_fulllength` and `minseqlength` set at 2. Sequentially we denoised and removed chimeras from read sequences with commands `cluster_unoise`, and `uchime3_denovo`, respectively (see Supplementary YYY for details). Finally, we clustered zOTUs at a 95% sequence similarity with command `usearch_global` and option `id` set at 0.95. After which, 3,035,960 sequence reads from experimental samples remained.

Taxonomy was assigned with the Tree-Based Alignment Selector Toolkit [v2.2; @carbone2019] by placing unknowns within the Pezizomycotina v2 reference tree [@carbone2017]. ITS sequences were blasted against the UNITE database by the ribosomal database project (RDP) classifier. A total of 2147 OTUs hits were obtained and are composed of 68.6% Ascomycota, 26.8% Basidiomycota,\<0.05% Chytridiomycota, \<0.05% Glomeromycota, \<0.05% Mortierellomycota, \<0.05% Rozellomycota, 0.05% Kickxellomycota, and 4.2 % BLAST hit misses. Only OTUs representing Ascomycota were used for downstream statistical analyses since foliar endophyte communities in tropical trees are dominated by Ascomycota [@arnold2007].

For each OTU identified, we removed laboratory contaminants from experimental samples by substracting the average read count found in control samples from the DNA extraction and PCR steps. Our analysis of mock communities allowed use to identify and remove false OTUs from experimental samples, those with fewer than 10 reads, and remove 0.1% of the read relative abundance across all samples [@oita2021]. Removed reads represent the frequency of reads classified as contamination in the mock communities relative to the expected read count. Three experimental samples from *Theobroma cacao* ($n$=2) and *Apeiba membranacea* ($n$=1) were removed from all analyses due to incomplete entries. After pruning taxa with zero reads from experimental samples, we identified 260 OTUs found exclusively in control ($E-$) plants ($n$=78) and deemed them as artifacts resulting from the greenhouse conditions. Consequently, these were consistently eliminated from treatment ($E+$) plants across all species. We converted reads for each fungal OTU to proportions of total sequence abundance per sample to reduce differences in sampling effort, following previous studies (@weiss2017; @mcmurdie2014 ). We then removed singletons and obtained an average of 2,464,558 sequence reads in 529 Ascomycota OTUs across 156 experimental samples of 7 tree species. All analyses post taxonomic assignment were performed in R [v. 4.3.2; @rcoreteam2023] using the `phyloseq` package [@mcmurdie2013] and custom scripts (see Supplementary Material).

### Ant-endophyte interaction assays

A fresh fourth leaf was used in ant assays. To assess leaf-cutter ant damage, we introduced one detached leaf per plant per treatment to an actively foraging leaf-cutter ant colony for a two-hour assay. We presented leaf-cutter ant colonies with a choice of an E+ or an E- leaf on one disposable plastic plate next to an active nest trail. Carefully, we collected and placed debris from the trail leading up to the plate to lure ants into the plate. We initiated the ant assay as soon as one ant entered the plate and explored the leaf contents (for \~ 10-20 seconds). Every five minutes we took a digital photo of the choice arena until about 75% of the leaf content of one of the leaves was consumed. We used the digital photo at time zero and at the end of trial to quantify the leaf area removed using ImageJ [v1.52r; @schneider2012]. Ant recruitment was estimated by counting individuals in the choice arena throughout trial event.

### Pathogen assays

For the pathogen assays, we introduced an agar plug inoculated with hyphae of *Calonectria* sp. ($P+$ treatment), and an agar plug without the pathogen ($P-$ control) to similarly aged/sized leaves within 10-14 days after endophyte inoculations (CITATION). Leaves with the $P+$ or $P-$ treatment were misted with sterile water two times a day (morning and afternoon) to maintain moisture. After four days, we removed the plugs and took digital photos to analyze leaf area damage using ImageJ [v1.52r; @schneider2012].

### Statistical Analyses

We explored how leaf functional traits and foliar fungal symbionts correlated to herbivory and pathogen damage on leaves. We present the analyses at the leaf and at the plant level. Leaf functional traits were measured and are presented in their raw form, at the leaf level, while FEF data was explored and is presented at the plant level. In analyses where leaf functional traits and FEF are combined we used averages of the leaf functional traits.

To test for H2, we used a general linear mixed model (GLM) with XXXX as the response variable. To determine which fixed effects to include in the models we used the `vif` function in *R* to calculate the variance inflation factor for all explanatory variables (ACI, LT, LPS and LMA) [@rcoreteam2023]. We then created a correlation matrix with `cor` function to assess correlations among covariates. We opted to maintain explanatory variables pertaining to physical barriers (LT, LPS and LMA) and exclude ACI from subsequent linear models due to high collinearity with LPS (0.54) and LMA (0.73). Every variable kept exhibits some degree of collinearity and this is well recorded in the literature (CITE HERE).

Additionally, Principal Component Analysis (PCA) was used to reduce dimensions among covariates and reveal underlying interactions that could influence fungal endophyte abundance, diversity and community composition in seedlings. The PCA was computed using the `prcomp` function in R statistical software [@rcoreteam2023]. A complete PCA was computed with variables ACI, LT, LPS, and LMA (FIGURE 2a?). We then proceeded to compute a PCA with the data from leaves of plants used in the ant ($n$ = 210) and pathogen assays ($n$ = 192).

The PCA revealed how covariates (LMA, ACI, Thickness and Toughness) interact. I overlapped tree species groups on the PCA axes to show how the variance in the data is explained by PC1 (60%) and PC2 (27%) (Fig. 4). This is indicative of correlation among covariates. Thickness and toughness were orthogonal to each other in PCA, indicative of low correlation.

# Results

Seedlings exposed to forest spore fall, *E+*, had a significantly higher proportion of leaf segments colonized by fungal endophytes across all species (data from cultures, Fig. S1) Using our molecular data set we saw that seedlings with *E+* treatment had a significantly higher fungal endophyte relative abundance (*p* = \<0.05) for all tree species when compared to *E-* treatment with the exception of *C. cainito* and *H. concinna* (Fig. 1). This reflects the successful inoculation of our treatment types. Despite these significant differences, there is a high degree of variability in endophyte relative abundance within each treatment type (Fig. 1).

For individuals of all tree species, we observed general differences in leaf functional traits summarized in Table 1. For LT we observed significant differences for *A. membranacea* , *Dypteryx* sp., and *L. panamensis* despite treatment type. Anthocyanins (ACI) were all leaf functional traits across all tree species. We did not see (CONTINUE HERE)

# Discussion

# References

::: {#refs}
:::

# Figures

```{r, Packages}
#| echo: false
#| eval: true
#| warning: false
# Data manipulation and visualizations ####
library("rmarkdown")
library("conflicted")
library("devtools")
#library("DiagrammeR")
library("factoextra")
library("ggfortify")
library("ggiraph")
library("ggiraphExtra")
library("ggplot2")
library("ggpubr")
library("gridExtra")
library("ggthemes")
library("ggraph")
library("hrbrthemes")
library("igraph")
library("tidyverse")
library("plotrix")
#library("viridis")
library("MetBrewer")
library("formatR")
library("wesanderson")
#library("knitr")
#library("rmarkdown")
library("data.table")
library("kableExtra")
library("janitor")
library("flextable")
library("vtable")
library("gtsummary")
library("officer")


# Stats ####
library("broom")
library("vegan")
library("nlme")
library("MASS")
library("car") #For type 3 Anova with Anova() function.
library("PerformanceAnalytics")
library("corrr")
library("permute")
library("lattice")
library("pls")
library("buildmer")
library("lme4")
library("sjPlot")
library("glmmTMB")
library("indicspecies")

# Phylogenetic analyses #source("https://raw.githubusercontent.com/joey711/phyloseq/master/inst/scripts/installer.R",local = TRUE)
library("mgcv")
library("phyloseq")
library("ape")
library("metagMisc")
#remotes::install_github("vmikk/metagMisc")
#library("speedyseq")
library("picante")
#source("http://bioconductor.org/biocLite.R")


# Preference of conflicted functions among packages ####
conflict_prefer("select", "dplyr")
conflict_prefer("filter", "dplyr")
conflict_prefer_all("phyloseq")
```

```{r, Clean sequences}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

#Working directory
rm(list = ls()) 
#setwd("H:/.shortcut-targets-by-id/0B9v0CdUUCqU5VVR4a3BvNHM1Z28/VBL_users/Grad_Students/Bolivar/Dissertation/Leaf_Traits_Panama/Data/Sequence_analyses")

#Working Directory and Path for Windows
#setwd("C:/Users/boloq/Box/Dissertation/Leaf_Traits_Panama/Data/Aim3_Sequence_analyses")
#save.path <- "C:/Users/boloq/Box/Dissertation/Leaf_Traits_Panama/Data/Aim3_Sequence_analyses"

#Working Directory and Path for Linux
setwd("/home/baponte/Boxx/Dissertation/Leaf_Traits_Panama/Data/Aim3_Sequence_analyses/")
# 
save.path <- ("/home/baponte/Boxx/Dissertation/Leaf_Traits_Panama/Data/Aim3_Sequence_analyses/")



set.seed(123)
# Cleaned (decontaminated) sample sequences ####
#csamp
# csamp <- read.csv("clean_data/otu_data/all_cleaned_at_10_percent.csv")
# csamp[,2:157] <- lapply(csamp[,2:157], as.integer)
# csamp <- csamp %>%
#   rename(OTU_ID = X) %>%
#   column_to_rownames(var = "OTU_ID") %>%
#   na.omit() %>%
#   na.fail() %>%
#   as.matrix()

# Saving
# saveRDS(csamp, file.path(save.path, "R_objects", "clean_sequences.rds"))

# Reading
csamp <- readRDS(file.path(save.path, "R_objects", "clean_sequences.rds"))
```

```{r, TBAS}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

# T-BAS taxonomic assignment ####
#tbas
# tbas <- read.csv("unite_report2BSDHQZP.csv")
# tbas <- tbas %>%
#   dplyr::filter(phylum == "Ascomycota") %>%
#   select(-match,-e.value, -percent, -coverage, -bitscore) %>% #eliminating some columns that I am not really using
#   na.omit()%>%
#   rename( OTU_ID = query) %>%
#   column_to_rownames(var = "OTU_ID") %>%
#   na.omit() %>%
#   na.fail() %>%
#   as.matrix()

# Saving
#saveRDS(tbas, file.path(save.path, "R_objects", "tbas_filtered.rds"))

# Reading
tbas <- readRDS(file.path(save.path, "R_objects", "tbas_filtered.rds"))
```

```{r, Leaf trait variables data}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

data <- read.csv("field_data/Aim3_Bolivar_Summer_2019_Leaf_Traits_datasheet.csv")

# data2
# Data set created on May 23, 2023 for analysis with raw data.
# data2 <- data %>% 
#   separate(Unique_ID, c("Trial_Code", "ELoad", "Species2", "Replicate")) %>%
#   unite("Sample_name", Species2:Replicate, sep = "") %>%
#   relocate(Sample_name, Trial_Code, E_load, .after = Species) %>%
#   select(!c(5, 15:17, 20)) %>%
#   rename( Anthocyanins = Anthocyanins..aci., 
#           Thickness = Thickness..micro.m.,  
#           Toughness =Toughness..lbf., 
#           LMA = LMA_dw, 
#           Abun_3 = Abundance_Proportion_day3, 
#           Abun_7 = Abundance_Proportion_day7)

# Saving
# saveRDS(data2, file.path(save.path, "R_objects", "leaf_data_raw_analyses.rds"))

# Reading
data2 <- readRDS(file.path(save.path, "R_objects", "leaf_data_raw_analyses.rds"))

# ndata (new_data)
# This data set can also be used to conduct analysis with "raw" data . It is summarized by leaf.

# ndata <- data %>%  
#   group_by(Species, Unique_ID, Trial_type, E_load, Leaf_ID) %>%
#   summarise(Anthocyanins..aci. = mean(Anthocyanins..aci., na.rm = TRUE), 
#             Thickness..micro.m. = mean(Thickness..micro.m., na.rm = TRUE),
#             Toughness..lbf. = mean(Toughness..lbf., na.rm = TRUE),
#             LMA_dw = mean(LMA_dw, na.rm = TRUE),
#             Abundance_Proportion_day3 = mean(Abundance_Proportion_day3, na.rm = TRUE),
#             Abundance_Proportion_day7 = mean(Abundance_Proportion_day7, na.rm = TRUE)) %>%
#    separate(Unique_ID, c("Trial_Code", "ELoad", "Species2", "Replicate")) %>%
#   unite("Sample_name", Species2:Replicate, sep = "") %>%
#   relocate(Species, Trial_Code,.after = Sample_name) %>%
#   rename( Anthocyanins = Anthocyanins..aci., 
#           Thickness = Thickness..micro.m.,  
#           Toughness =Toughness..lbf., 
#           LMA = LMA_dw, 
#           Abun_3 = Abundance_Proportion_day3, 
#           Abun_7 = Abundance_Proportion_day7) %>%
#   select(!ELoad) %>%
#   ungroup() %>%
#    mutate_if(is.character, as.factor) %>%
#   na.omit() %>% 
#   na.fail()

# Saving
# saveRDS(ndata, file.path(save.path, "R_objects", "new_leaf_data.rds"))

# Reading
ndata <- readRDS(file.path(save.path, "R_objects", "new_leaf_data.rds"))

# Splitting and relocating Unique_ID column 
# Extract names to match to phyloseq objects separate(Species2, c("Species3", "Numbers"), sep = "(?<=[A-Za-z])(?=[0-9])")

# ndata2 (new_data2)
# Data frame is summarized by plant (n = 156)

# ndata2 <- ndata %>% 
#   select(!c(Abun_3, Abun_7)) %>%
#   mutate_if(is.character, as.factor) %>%
#   group_by(Sample_name, Species, Trial_type, Trial_Code, E_load) %>%
#   summarise(Anthocyanins = mean(Anthocyanins, na.rm = TRUE),
#             Thickness = mean(Thickness, na.rm = TRUE),
#             Toughness = mean(Toughness, na.rm =TRUE),
#             LMA = mean(LMA, na.rm = TRUE)) 

# Saving
# saveRDS(ndata2, file.path(save.path, "R_objects", "new_leaf_data2.rds"))

# Reading
ndata2 <- readRDS(file.path(save.path, "R_objects", "new_leaf_data2.rds"))
```

```{r, Ant assay data}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

#ants <- read.csv("field_data/Aim3_Bolivar_Summer_2019_Ant_Assay_datasheet.csv")
# 
# ants <- ants %>%
#   separate(Unique.ID, c("Trial_Code", "ELoad", "Species2", "Replicate")) %>%
#   unite("Sample_name", Species2:Replicate, sep = "") %>%
#   relocate(Species, Trial_Code,.after = Sample_name)%>%
#   slice(-c(11,12)) %>%
#   dplyr::filter(Trial_success!="0") %>%
#   select(!c(Leaf_ID, 
#             Colony_ID, 
#             Picture_ID, 
#             Colony_location_Pipeline.Gamboa, 
#             Attempt, 
#             Date_analyzed, 
#             X_coordinates, 
#             Y_coordinates, 
#             Elevation_meters, 
#             ELoad, 
#             Analyzer, 
#            
#             Time_start_trial, 
#             Time_first_contact, 
#             Time_first_cut, 
#             Time_end_elapsed_trial )) %>%
#   mutate(Ant_percent_leafloss = -100*{(Leaf_area_final-Leaf_area_initial)/Leaf_area_initial}) %>%
#   mutate_if(is.character, as.factor)

# Saving
# saveRDS(ants, file.path(save.path, "R_objects", "ants_data.rds"))

# Reading
ants <- readRDS(file.path(save.path, "R_objects", "ants_data.rds"))


# Just the area loss
ants2 <- ants %>%
  select(c(1,3,4,9,10))

ants3 <- ants %>%
  select(c(1,3,4,5,9,10))
```

```{r, Pathogen Assay Data}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

# patho <- read.csv("field_data/Aim3_Bolivar_Summer_2019_Patho_Assay_datasheet.csv")
# 
# patho <- patho %>%
#   separate(Unique.ID, c("Trial_Code", "ELoad", "Species2", "Replicate")) %>%
#   unite("Sample_name", Species2:Replicate, sep = "") %>%
#   relocate(Species, Trial_Code,.after = Sample_name) %>%
#   slice(-c(41:48, 97:100)) %>%
#   select(!c(ELoad,
#             Leaf_ID,
#             Trial_start_time,
#             Date, 
#             ImageJ_link, 
#             X, 
#             Notes)) %>%
#   rename(Treatment = Treatment_Ctrl_Pathogen) %>%
#   mutate(Pathogen_percent_damage = (Damage_area/Leaf_area)) %>%
#   mutate_if(is.character, as.factor)

# Saving
# saveRDS(patho, file.path(save.path, "R_objects", "pathogen_data.rds"))

# Reading
patho <- readRDS(file.path(save.path, "R_objects", "pathogen_data.rds"))


#Just the area damaged
patho2 <- patho %>%
  select(c(1,3,4,8,9,12))


patho3 <- patho %>%
  select(c(1,3,4,6,7,8,9,12))
```

```{r, Objects and dataframes}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

### Phyloseq objects ####

# OTU Table: 1774 OTU with 156 Samples
# OTU <- otu_table(csamp, taxa_are_rows = TRUE)
# OTU

# Saving
# saveRDS(OTU, file.path(save.path, "R_objects", "otu_table.rds"))

# Taxonomic Table: 1473 OTU by 7 taxonomic ranks
# TAX <- tax_table(tbas)

# Saving
# saveRDS(TAX, file.path(save.path, "R_objects", "taxonomy_table.rds"))

# SAMP (sample metadata)

# SAMP <- ndata2 %>%
#   column_to_rownames(var = "Sample_name")
# SAMP <- sample_data(SAMP)

# Saving
# saveRDS(SAMP, file.path(save.path, "R_objects", "samp_data.rds"))


# Merging phyloseq objects ####
# pq1 <- phyloseq(OTU, TAX, SAMP)

# Saving
# saveRDS(pq1, file.path(save.path, "R_objects", "phyloseq_main.rds"))

# Reading
pq1 <- readRDS(file.path(save.path, "R_objects", "phyloseq_main.rds"))

# Random tree ####
#random_tree <- rtree(ntaxa(pq), rooted=TRUE, tip.label=taxa_names(pq))

```

```{r, Mareli code}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

# pq2 <- prune_taxa(taxa_sums(pq1) > 0, pq1)
# ntaxa(pq2) #There are 488 taxa with 0 reads.

# Saving
# saveRDS(pq2, file.path(save.path, "R_objects", "phyloseq_pq2.rds"))

# Reading
pq2 <- readRDS(file.path(save.path, "R_objects", "phyloseq_pq2.rds"))

# Subset of pq1 -- only E+ samples and OTUs.
#Important -- don't remove OTUs with 0 reads here, because we want to know which ones have no reads in E+ but DO have reads in E-.
pq2E <- subset_samples(pq2, E_load == "E+")
#ntaxa(pq2E) #1219 taxa and 78 samples

# Eliminate E- only OTUs. Keep OTUs with more than 0 reads in subset.
pq2_EminusRemoved <- prune_taxa(taxa_sums(pq2E) > 0, pq2E)
#ntaxa(pq2_EminusRemoved) #959 taxa remain. 260 OTUs removed.

#If OTU has 0 reads in E+, it will have reads in E-, so this prunes pq2E and identifies all OTUs that have 0 reads in E+, which by default, are the OTUs that DO have reads in E-.
pq2_EminuOnly <- prune_taxa(taxa_sums(pq2E) == 0, pq2E)
#pq2_EminuOnly #260 taxa and 78 samples

# Gets names of OTUs and make a data frame with column name the same as in pq2
EminusOTUs2 <- taxa_names(pq2_EminuOnly)
EminusOTUsList2 <- as.data.frame(EminusOTUs2)
#write.table(EminusOTUsList2, "clean_data/otu_data/Aim3_RemovedOTUsList.txt")

# Get vectors (names) of numbered OTUs from pq1
OTU_ID <- rownames(otu_table(pq1))

# Returns all OTU names that don't have a match in Eminus OTUs.
notShared <- setdiff(OTU_ID, EminusOTUs2)

# Subset phyloseq object to only these OTUs
pq2ET <- subset(otu_table(pq2), rownames(otu_table(pq2)) %in% notShared)

# Exported into "Sequence Analyses"
#write.csv(pq2ET, "clean_data/Aim3_OTU_phyloseq_trimmed.csv")

# New phyloseq object without E- OTUs.
newpq <- merge_phyloseq(pq2ET, tax_table(pq2), sample_data(pq2), rtree(ntaxa(pq2), rooted=TRUE, tip.label=taxa_names(pq2)))

# Relative Abundance of 959 OTUs. No removal of singletons yet.

# rawra <- transform_sample_counts(newpq, function(x)x/sum(x))
# rawRA <- subset(otu_table(rawra), rownames(otu_table(rawra)) %in% notShared)
# write.csv(rawRA, "clean_data/otu_data/Aim3_OTU_Relative_Abundance_singletons_untrimmed.csv")

# Outputs of this script -- OTU table without E minus only OTUs and phyloseq object with E- OTUs removed. No removal of singletons yet.

# End of Mareli's Code #
```

```{r, Singleton removal}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

# Filtering Taxa: Removal of singletons ####
 # Removal of  singletons
newpq1 <- filter_taxa(newpq, function (x) {sum(x > 0) > 1}, prune=TRUE)
# The result is 2,464,558 sequence reads from 569 taxa in 156 samples.

# Relative abundance calculation ####
# Code below is for making the .csv file. After that refer to df rawra from .csv file.

# newpqRA <- transform_sample_counts(newpq1, function(x)x/sum(x))
# rabun <- subset(otu_table(newpqRA), rownames(otu_table(newpqRA)) %in% notShared)
# write.csv(rabun, "clean_data/otu_data/Aim3_OTU_Relative_Abundance_singletons_trimmed.csv")
#GPfr = filter_taxa(GPr, function(x) mean(x) > 1e-5, TRUE)


# Richness data frame ####
rich <- estimate_richness(newpq1, split = TRUE, measures = c("Observed","Shannon"))

rich <- rich %>%
  rownames_to_column() %>%
  rename(Sample_name = rowname)

# Phyloseq object to data frames ####

# Cleaned: no singletons
newpqDF <- phyloseq_to_df(newpq1, addtax = T, addtot = F, addmaxrank = F, sorting = "abundance")
#write.csv(newpqDF, "clean_data/otu_data/Aim3_OTUs_assigned_nonsingletons.csv")
```

```{r, Loading phyloseq products}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

#These are the data frames resulting from the phyloseq data wrangling.

#The 569 OTUs and 156 samples (total abundances)
newpqDF <- read.csv("clean_data/otu_data/Aim3_OTUs_assigned_nonsingletons.csv")
newpqDF <- newpqDF %>%  # Code from the Taxonomical assginment section dopes the same. This is just calling it directly from the /csv to avoid running all the code. 
  select(!c(1))

# OTUs 959 and 156 samples (total abundances)

# eOTU <- read.csv("clean_data/Aim3_OTU_phyloseq_trimmed.csv") 
# Can upload as data frame. Use for distance matrices.
# 
# eOTU2 <- eOTU %>%
#   rename(OTU_ID = X) %>%
#   column_to_rownames(var = "OTU_ID") %>%
#   t() %>%
#   as.data.frame() %>%
#   rownames_to_column() %>%
#   rename(Sample_name = rowname)

# Saving
#saveRDS(eOTU2, file.path(save.path, "R_objects", "eOTU2.rds"))

# Reading
eOTU2 <- readRDS(file.path(save.path, "R_objects", "eOTU2.rds"))


# Just 569 OTUs and 156 samples (total abundances)
# cOTU <- newpqDF %>%
#   select(-kingdom, -phylum, -class, -order, -family, -genus, -species) %>%
#   column_to_rownames(var = "OTU") %>%
#   as.matrix() %>%
#   t() %>%
#   as.data.frame() %>%
#   rownames_to_column() %>%
#   rename(Sample_name = rowname)

# Saving
# saveRDS(cOTU, file.path(save.path, "R_objects", "clean_OTU.rds"))

# Reading
cOTU <- readRDS(file.path(save.path, "R_objects", "clean_OTU.rds"))

# Relative Abundance matrix (trimmed to 569 OTUs)
# rabun <- read.csv("clean_data/otu_data/Aim3_OTU_Relative_Abundance_singletons_trimmed.csv")
# rabun <- rabun %>%
#   rename(OTU_ID = X) %>%
#   column_to_rownames(var = "OTU_ID") %>%
#   na.omit() %>%
#   na.fail() %>%
#   as.matrix()

# Saving
# saveRDS(rabun, file.path(save.path, "R_objects", "rabun_tidy.rds"))

# Reading
rabun <- readRDS(file.path(save.path, "R_objects", "rabun_tidy.rds"))

# Relative Abundance matrix (singletons untrimmed Ascomycota 959 OTUs)
# rawra <- read.csv("clean_data/otu_data/Aim3_OTU_Relative_Abundance_singletons_untrimmed.csv")
# rawra <- rawra %>%
#   rename(OTU_ID = X) %>%
#   column_to_rownames(var = "OTU_ID") %>%
#   na.omit() %>%
#   na.fail() %>%
#   as.matrix()

# Saving
# saveRDS(rawra, file.path(save.path, "R_objects", "rawra_tidy.rds"))

# Reading
rawra <- readRDS(file.path(save.path, "R_objects", "rawra_tidy.rds"))

```

## Figure 1

```{r, Master data set}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

#This contains all the OTU id's per sample type. Hence, 156 (# samples) observations per OTU_ID. The phyloseq object achieves the same purpose. This data set is easier to use with other functions or test not supported by phyloseq.

# Master ####
#put all data frames into list
df_list_ant <- list(ndata2, cOTU, ants2, patho2, rich)    

#merge all data frames together
master <- df_list_ant %>% 
  reduce(left_join, by= "Sample_name", "E_load") %>%
  mutate(Total_reads = sum(across(starts_with("OTU")))) %>%
  group_by(Species) %>%
  mutate(Relative_Abundance = Total_reads/sum(Total_reads)) %>%
  select(c(1:9, 579:591)) %>%
  select(!c(E_load,
            E_load.y,
            Trial_Code,
            Trial_Code.y)) %>%
  relocate(Treatment, E_load.x, .after = Trial_type) %>%
  rename(Pathogen_Damage_area = Damage_area, 
         E_load = E_load.x, 
         Trial_Code = Trial_Code.x)
```

```{r, Ant Master Data}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

# Masterant on a per plant level
masterant <- master %>%
  subset(Trial_type == "Ant") %>%
  drop_na(Trial_Code) %>%
  select(!c(4, 13:14)) %>%
  na.omit()

# Masterant on a per leaf level (70 plants, 3 leaves per plant = 210 leaves measured)
df_list_ant2 <- list(ndata, cOTU, ants2, patho2, rich)

masterant2 <- df_list_ant2 %>%
  reduce(left_join, by= "Sample_name", "E_load") %>%
 mutate(Total_reads = sum(across(starts_with("OTU")))) %>%
  group_by(Species) %>%
  mutate(Relative_Abundance = Total_reads/sum(Total_reads)) %>%
  select(c(1:10, 584:594)) %>%
  select(!c(E_load,
            Trial_Code)) %>%
  relocate(Treatment, E_load.x, .after = Trial_type) %>%
  rename(Pathogen_Damage_area = Damage_area, 
         E_load = E_load.x, 
         Trial_Code = Trial_Code.x) %>%
  subset(Trial_type == "Ant") %>%
  drop_na(Trial_Code) %>%
  select(!c(5, 14:15)) %>%
  na.omit() %>%
  mutate(logit_herbivory = logit(Ant_percent_leafloss))
```

```{r, Pathogen Master data}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false


# Pathogen Observations per plant level
masterpat <- master %>%
  subset(Trial_type == "Pathogen") %>%
  select(!c(11,12)) %>%
  na.omit()

# Pathogen Observations per leaf level (64 plants, 3 leaves per plant = 192 leaves measured)
# masterpat2

masterpat2 <- df_list_ant2 %>%
  reduce(left_join, by= "Sample_name", "E_load") %>%
 mutate(Total_reads = sum(across(starts_with("OTU")))) %>%
  group_by(Species) %>%
  mutate(Relative_Abundance = Total_reads/sum(Total_reads)) %>% 
  select(c(1:10, 584:594)) %>%
  select(!c(E_load,
            Trial_Code)) %>%
  relocate(Treatment, .after = Trial_type) %>%
  rename(Pathogen_Damage_area = Damage_area, 
         E_load = E_load.x, 
         Trial_Code = Trial_Code.x) %>%
  subset(Trial_type == "Pathogen") %>%
  select(!c(12:13)) %>%
  na.omit() %>%
  mutate(logit_pathogenicity = logit(Pathogen_percent_damage)) 

#mutate(Pathogen_percent_damage = pmin(pmax(Pathogen_percent_damage, 0), 1)) %>%  # Rescale values between 0 and 1
  #mutate(logit_damage = logit(Pathogen_percent_damage))
#masterpat2$log_damage <- log(masterpat2$Pathogen_percent_damage)
```

```{r fig1}
#| echo: false
#| eval: true
#| warning: false
#| fig.width: 7
#| fig.height: 7
#| fig.cap: "Relative abundance (RA) of Ascomycota OTUs of seven tree species used in the study. Violin plots show the distribution of the RA. The horizontal line within the embedded boxplots represents the median, the box represents the interquartile range (IQR), and the whiskers represent the 1.5xIQR. Outliers are represented by dots. Pink fille violin plots represent low endophyte (*E-*) treatment and yellow fille violins represent high endophyte (*E+*) treatment. Relative abundance is the percentage of endophyte colonization within individuals of the same species. Significance levels are represented by asterisks [*p* = 0.05 (\\*), *p* = 0.01 (\\**), and *p* = 0.001 (\\***)]."

library("ggplot2")
library("ggpubr")

# RA per E_load

newname <- c("A. membranacea", "C. cainito", "C. alliodora", "Dypterix .sp", "H. concinna", "L. panamensis", "T. cacao")

names(newname) <- c("APEIME", "CHRYCA", "CORDAL", "DYPTE", "HEISCO", "LACPA", "THEOCA")

RAE <- ggplot(data= master, aes(x = E_load, y = Relative_Abundance)) +
  geom_violin(aes(fill = E_load)) +
    geom_boxplot(width=0.1, color="grey40", alpha=0.2, outlier.shape = -1) +
# geom_crossbar(stat="summary", fun = mean, fun.max = mean, fun.min = mean,  fatten = 2, width = 0.5) +
# geom_point(color="black", alpha = 0.4,size = 2, position = position_jitter(w=0.05)) +
  scale_fill_manual(values= met.brewer("Cross", n = 2, type = "discrete")) +
  scale_color_manual(values= met.brewer("Cross", n = 2, type = "discrete")) +
  theme_bw(base_size = 14)+
  
  theme(panel.grid.major = element_line(colour = "grey")) +
  theme(panel.grid.minor = element_line(colour = "grey")) +
  theme(strip.background = element_rect(fill = "white")) +
  theme(strip.text.x = element_text(colour = "grey15", face = "italic")) +
  theme(legend.title = element_text(face = "bold")) +
  theme(panel.border = element_rect(linetype = "blank", fill = NA)) +
  labs(x="", y="Relative Abundance", face = "bold")+
  theme(legend.position="right", legend.title = element_blank(),
        legend.text = element_text(size = 16),
        axis.text.x=element_blank()) +
  #labs(caption = "*Outliers and singletons removed.") +
  stat_compare_means(comparisons = list(c("E-", "E+")), 
                     method = "t.test",
                     label = "p.signif", label.y = 0.3, bracket.size = 0.65) +
  facet_wrap(~Species, labeller = as_labeller(newname))
  

RAE
#ggsave(filename="./plots/Aim3_RA_Eload_SEPT182023.png", plot = RAE, dpi=600, units=c("mm"), width = 180, height = 180)
```

## Figure 2

```{r, Full PCA}
#| echo: false
#| eval: true
#| warning: false
#| tidy: true

# PCA using covariates to explain species richness/abundance ####
data.pca <- ndata[c(7,8,9,10)]
data.pca <- data.pca %>%
  rename(ACI = Anthocyanins, LT = Thickness, LPS = Toughness) #Renaming columns for plotting purposes.

###Run this to create pca with prcomp function
pca<-prcomp(data.pca, scale=TRUE)
pca$rotation=-pca$rotation
pca$x=-pca$x

#Checking the PCA
#plot(pca,type = "lines")
#biplot(pca) # Base type PCA

# PCA using autoplot() and prcomp()and modifying with ggplot syntax ####
auto <- autoplot((pca), data = data.frame(ndata), # For some reason this functions require data to be data= data.frame()
         alpha=0, #Setting alpha to zero render the automatic circle point null.Manipulate shapes with geom_point(). 
         loadings = TRUE,loadings.colour = "black",
         loadings.label=TRUE, loadings.label.colour="black",
         loadings.label.size= 4, size = 7, loadings.label.vjust = 1, 
         loadings.label.hjust = 1) + 
  geom_point(aes(fill = Species, color = Species, shape = E_load), alpha = 0.6, size = 3) +
  geom_hline(yintercept = 0, colour = "gray45") +
  geom_vline(xintercept = 0, colour = "gray45") +
  scale_color_manual(labels=c("A. membranacea", "C. cainito", "C. alliodora", "Dypterix .sp", "H. concinna", "L. panamensis", "T. cacao"),values = met.brewer(name = "Cross", n = 8, type="discrete")) +
  stat_ellipse(aes(color=Species), geom = "path", size = 1, position = "identity", type = "t", linetype = 1, level = 0.95, segments = 51, na.rm = FALSE, show.legend = NA, inherit.aes = TRUE) +
  scale_fill_discrete(labels = c("A. membranacea", "C. cainito", "C. alliodora", "Dypterix .sp", "H. concinna", "L. panamensis", "T. cacao")) +
  theme_classic(base_size = 16) +
  #labs(title = "(a)") + 
  theme(plot.title = element_text(color="black", size = 14, face="bold")) +
  theme(legend.text = element_text(color="black", face="italic"))

#caption = expression("ACI = anthocyanins, LPS = leaf punch strength, LMA = leaf mass per area, LT = leaf thickness. All leaf replicates per species ("~ italic("n") ~ "= 467).") #This goes in the labs() arguments

auto$layers <- c(auto$layers, auto$layers[[2]], auto$layers[[3]]) # This adds/copies layers 2-3 and overlays them. It makes the arrows be on top of the points. There must be a better ways of doing this.

#Checking if all is good.
auto <- auto +
  theme(plot.caption = element_text(size = 6, hjust = 1, vjust = 1))
#auto 

#ggsave(filename="./plots/Aim3_PCA_SEPT252023.png", plot = auto, dpi=600, units=c("mm"), width = 180, height = 180, bg = "white")
```

```{r, Ant PCA, tidy=TRUE}
#| echo: false
#| eval: true
#| warning: false

# PCA using covariates to explain species richness/abundance ####
ant_pcadata <- masterant2[c(7,8,9,10)]
ant_pcadata <- ant_pcadata %>%
  rename(ACI = Anthocyanins, LT = Thickness, LPS = Toughness) #Renaming columns for plotting purposes.


###Run this to create pca with prcomp function
ant.pca <-prcomp(ant_pcadata, scale=TRUE)
ant.pca$rotation=-ant.pca$rotation
ant.pca$x=-ant.pca$x

# Plot
auto_ant <- autoplot((ant.pca), data = data.frame(masterant2), # For some reason this functions require data to be data= data.frame()
         alpha=0, #Setting alpha to zero render the automatic circle point null.Manipulate shapes with geom_point(). 
         loadings = TRUE,loadings.colour = "black",
         loadings.label=TRUE, loadings.label.colour="black",
         loadings.label.size= 4, size = 7, loadings.label.vjust = 0, 
         loadings.label.hjust = 1.2) + 
  geom_point(aes(fill = Species, color = Species, shape = E_load), alpha = 0.6, size = 3) +
  geom_hline(yintercept = 0, colour = "gray45") +
  geom_vline(xintercept = 0, colour = "gray45") +
  scale_color_manual(values = met.brewer(name = "Cross", n = 8, type="discrete")) +
  stat_ellipse(aes(color=Species), geom = "path", size = 1, position = "identity", type = "t", linetype = 1, level = 0.95, segments = 51, na.rm = FALSE, show.legend = NA, inherit.aes = TRUE) +
  theme_classic(base_size = 12) +
  #labs(title = "(b)") +
  theme(plot.title = element_text(color="black", size = 14, face="bold")) +
  guides(color = "none")  +      #To turn off color legend
  guides(fill = "none") + #To turn off fill legend
  guides(shape = "none")
# caption = expression("ACI = anthocyanins, LPS = leaf punch strength, LMA = leaf mass per area, LT = leaf thickness. All leaf replicates per species ("~ italic("n") ~ "= 210).")
  
  


auto_ant$layers <- c(auto_ant$layers, auto_ant$layers[[2]], auto_ant$layers[[3]]) # This adds/copies layers 2-3 and overlays them. It makes the arrows be on top of the points. There must be a better ways of doing this.

#Checking if all is good.
auto_ant <- auto_ant +
  theme(plot.caption = element_text(size = 6, hjust = 1, vjust = 1))
#auto_ant

#ggsave(filename="./plots/Aim3_Ant_PCA_07102023.png", plot = auto_ant, dpi=600, units=c("mm"), width = 180, height = 180, bg = "white")
```

```{r, Pathogen PCA, tidy=TRUE}
#| echo: false
#| eval: true
#| warning: false
#
# PCA using covariates to explain species richness/abundance ####
pat_pcadata <- masterpat2[c(8,9,10,11)]
pat_pcadata <- pat_pcadata %>%
  rename(ACI = Anthocyanins, LT = Thickness, LPS = Toughness) #Renaming columns for plotting purposes.


###Run this to create pca with prcomp function
pat.pca <-prcomp(pat_pcadata, scale=TRUE)
pat.pca$rotation=-pat.pca$rotation
pat.pca$x=-pat.pca$x

# Plot
auto_pat <- autoplot((pat.pca), data = data.frame(masterpat2), # For some reason this functions require data to be data= data.frame()
         alpha=0, #Setting alpha to zero render the automatic circle point null.Manipulate shapes with geom_point(). 
         loadings = TRUE,loadings.colour = "black",
         loadings.label=TRUE, loadings.label.colour="black",
         loadings.label.size= 4, size = 7, loadings.label.vjust = 0, 
         loadings.label.hjust = 1.2) + 
  geom_point(aes(fill = Species, color = Species, shape = E_load), alpha = 0.6, size = 3) +
  geom_hline(yintercept = 0, colour = "gray45") +
  geom_vline(xintercept = 0, colour = "gray45") +
  scale_color_manual(values = met.brewer(name = "Cross", n = 8, type="discrete")) +
  stat_ellipse(aes(color=Species), geom = "path", size = 1, position = "identity", type = "t", linetype = 1, level = 0.95, segments = 51, na.rm = FALSE, show.legend = NA, inherit.aes = TRUE) +
  theme_classic(base_size = 12) +
  #labs(title = "(c)") +
  theme(plot.title = element_text(color="black", size = 14, face="bold")) + 
  guides(color = "none")  +      #To turn off color legend
  guides(fill = "none") + #To turn off fill legend
  guides(shape = "none")
# caption = expression("ACI = anthocyanins, LPS = leaf punch strength, LMA = leaf mass per area, LT = leaf thickness. All leaf replicates per species ("~ italic("n") ~ "= 210).")
  

auto_pat$layers <- c(auto_pat$layers, auto_pat$layers[[2]], auto_pat$layers[[3]]) # This adds/copies layers 2-3 and overlays them. It makes the arrows be on top of the points. There must be a better ways of doing this.

#Checking if all is good.
auto_pat <- auto_pat +
  theme(plot.caption = element_text(size = 6, hjust = 1, vjust = 1))
#auto_pat

#ggsave(filename="./plots/Aim3_Pathogen_PCA_08162023.png", plot = auto_pat, dpi=600, units=c("mm"), width = 180, height = 180, bg = "white")
```

```{r, Fig2 Grid PCA}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false
#| fig.width:  10
#| fig.height: 10
#| fig.cap: "Leaf functional traits are conserved within tree species regardles of endophyte load treatment. (a) Principal Component Analysis (PCA) of leaf functional traits from all tree species separated by *E-* and *E+* treatment. (b) PCA of leaf functional traits of plants solely used in ant herbivory assays. (c) PCA leaf functional traits of plants used solely in pathogen damage assays. Colors represent individual species. Circle and triangles represent los (*E-*) and high (*E+*) endophyte treatments, respectively. Colored ellipses correspond to tree species and represent 95% confidence intervals."
grid_pca <- ggarrange(auto, 
                      ggarrange(auto_ant, auto_pat, nrow = 1, ncol=2, labels = c("(b)", "(c)")), nrow = 2, common.legend = TRUE, legend="bottom", labels = "(a)")

grid_pca
# Adding caption
# ant_caption <- expression("*All individual leaves per individual per tree species included ("~ italic("n") ~ "= 210).")
# ant_pca_with_caption <- annotate_figure(antpca_arranged, bottom = text_grob(ant_caption), fig.lab.pos = "bottom.right", fig.lab.size = 6)
# 
# ant_pca_with_caption

#ggsave(filename = "./plots/Aim3_GridPCA_SEPT182023.png", plot = grid_pca, dpi=600, units=c("mm"), width = 275, height = 300, bg = "white")
```

## Figure 3

```{r, Ant LM PCA data}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false

# PCA loadings (rotations) data frame
ant.pci<-data.frame(ant.pca$x) #Species = ndata$Species not included.

# joined data frame for PC llinear regressions

ant_pcalm <- masterant2[c(1,2,5,12, 17)]

ant_pcalm <- cbind(ant_pcalm, ant.pci) #Both data sets have 467 observations
```

```{r, Ant PC1-2}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false
# Formatting formula for p value
format.p <- function(p, precision = 0.001) {
  digits <- -log(precision, base = 10)
  p <- formatC(p, format = 'f', digits = digits)
  if (p < 0.001) {
    p = paste0('< ', precision)}
  if (p >= 0.001) {
    p = paste0('= ', p)    }
  sub("0", "", p)
}

lmpc1 <- lm(Ant_percent_leafloss ~ PC1, data = ant_pcalm)

# PC1
# Formatted p-value
p1 <- cor.test(ant_pcalm$Ant_percent_leafloss, ant_pcalm$PC1)$p.value

p1 = format.p(p1) # This is just to make the p-value nicer. 

## Plot PC1
       
ant_pc1 <- ggplot(ant_pcalm, aes(PC1, Ant_percent_leafloss)) +
  geom_jitter(aes(color = Species, shape = E_load), size = 2.2, alpha = 0.6) +
  geom_smooth(method = lm, se = T , level = 0.95, na.rm = F, color = "black", size = 0.75) +
  stat_regline_equation(aes(label = paste(..eq.label.., ..adj.rr.label.., sep = "~~~~"))) +
  #stat_regline_equation(label.y.npc = "top", label.x.npc = "center", aes(label = ..eq.label..), color= "black", size = 2.2) +
  scale_color_manual(labels=c("A. membranacea", "C. cainito", "C. alliodora", "Dypterix .sp", "H. concinna", "L. panamensis", "T. cacao"),values = met.brewer(name = "Cross", n = 8, type="discrete")) +
  theme_minimal(base_size = 12) +
  labs(title = "a", y = "Herbivory (%)", caption = "") +
  theme(plot.title = element_text(color="black", size = 14, face="bold")) + 
    annotate(geom="text",label=sprintf("italic('p')~'%s'", p1),parse=TRUE,x=-2.5,y=65) +
  theme(legend.text = element_text(color="black", face="italic"))
  #guides(color = "none")  +      #To turn off color legend
  #guides(fill = "none") + #To turn off fill legend
  #guides(shape = "none")

#ant_pc1

#ggsave(filename="./plots/Aim3_AntPC1_JUL102023.png", plot = ant_pc1, dpi=600, units=c("mm"), width = 180, height = 180, bg = "white")


# PC2
# Formatted p-values
p2 <- cor.test(ant_pcalm$Ant_percent_leafloss, ant_pcalm$PC2)$p.value

p2 =format.p(p2) # This is just to make the p-value nicer. 

## Plot PC2
lmpc2 <- lm(Ant_percent_leafloss ~ PC2, data = ant_pcalm)

ant_pc2 <- ggplot(ant_pcalm, aes(PC2, Ant_percent_leafloss)) +
  geom_jitter(aes(color = Species, shape = E_load), size = 2.2, alpha = 0.6) +
  geom_smooth(method = lm, se = T , level = 0.95, na.rm = F, color = "black", size = 0.75) +
  stat_regline_equation(aes(label = paste(..eq.label.., ..adj.rr.label.., sep = "~~~~"))) +
  #stat_regline_equation(label.y.npc = "top", label.x.npc = "center", aes(label = ..eq.label..), color= "black", size = 2.2) +
  scale_color_manual(values = met.brewer(name = "Cross", n = 8, type="discrete")) +
  theme_minimal(base_size = 12) +
  labs(title = "b", y = "Herbivory (%)", caption = "") +
  theme(plot.title = element_text(color="black", size = 14, face="bold")) +
 annotate(geom="text",label=sprintf("italic('p')~'%s'", p2),parse=TRUE,x=-1,y=65) +
    guides(color = "none")  +      #To turn off color legend
  guides(fill = "none") + #To turn off fill legend
  guides(shape = "none")
  
  
#ant_pc2

#ggsave(filename="./plots/Aim3_AntPC2_JUL102023.png", plot = ant_pc2, dpi=600, units=c("mm"), width = 180, height = 180, bg = "white")
```

```{r, Pathogen LM PCA data}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false
# PCA loadings (rotations) data frame
pat.pci<-data.frame(pat.pca$x) #Species = ndata$Species not included.

# joined data frame for PC llinear regressions

pat_pcalm <- masterpat2[c(1,2,5,6,13)]

pat_pcalm <- cbind(pat_pcalm, pat.pci) #Both data sets have 467 observations
```

```{r, Pathogen PC1-2}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false
# Formatting formula for p value
format.p <- function(p, precision = 0.001) {
  digits <- -log(precision, base = 10)
  p <- formatC(p, format = 'f', digits = digits)
  if (p < 0.001) {
    p = paste0('< ', precision)}
  if (p >= 0.001) {
    p = paste0('= ', p)    }
  sub("0", "", p)
}

plmpc1 <- lm(Pathogen_percent_damage ~ PC1, data = pat_pcalm)

# PC1
# Formatted p-value
p1 <- cor.test(pat_pcalm$Pathogen_percent_damage, pat_pcalm$PC1)$p.value

p1 = format.p(p1) # This is just to make the p-value nicer. 

## Plot PC1
       
pat_pc1 <- ggplot(pat_pcalm, aes(PC1, Pathogen_percent_damage)) +
  geom_jitter(aes(color = Species, shape = E_load), size = 2.2, alpha = 0.6) +
  geom_smooth(method = lm, se = T , level = 0.95, na.rm = F, color = "black", size = 0.75) +
  stat_regline_equation(aes(label = paste(..eq.label.., ..adj.rr.label.., sep = "~~~~"))) +
  #stat_regline_equation(label.y.npc = "top", label.x.npc = "center", aes(label = ..eq.label..), color= "black", size = 2.2) +
  scale_color_manual(values = met.brewer(name = "Cross", n = 8, type="discrete")) +
  theme_minimal(base_size = 12) +
  labs(title = "c", y = "Pathogen damage (%)", caption = "") +
  theme(plot.title = element_text(color="black", size = 14, face="bold"))+ 
    annotate(geom="text",label=sprintf("italic('p')~'%s'", p1),parse=TRUE,x=-2.5,y=0.75) +
  guides(color = "none")  +      #To turn off color legend
  guides(fill = "none") + #To turn off fill legend
  guides(shape = "none")

#pat_pc1

#ggsave(filename="./plots/Aim3_PatPC1_AUG282023.png", plot = pat_pc1, dpi=600, units=c("mm"), width = 180, height = 180, bg = "white")


# PC2
# Formatted p-values
p2 <- cor.test(pat_pcalm$Pathogen_percent_damage, pat_pcalm$PC2)$p.value

p2 =format.p(p2) # This is just to make the p-value nicer. 

## Plot PC2
plmpc2 <- lm(Pathogen_percent_damage ~ PC2, data = pat_pcalm)

pat_pc2 <- ggplot(pat_pcalm, aes(PC2, Pathogen_percent_damage)) +
  geom_jitter(aes(color = Species, shape = E_load), size = 2.2, alpha = 0.6) +
  geom_smooth(method = lm, se = T , level = 0.95, na.rm = F, color = "black", size = 0.75) +
  stat_regline_equation(aes(label = paste(..eq.label.., ..adj.rr.label.., sep = "~~~~"))) +
  #stat_regline_equation(label.y.npc = "top", label.x.npc = "center", aes(label = ..eq.label..), color= "black", size = 2.2) +
  scale_color_manual(values = met.brewer(name = "Cross", n = 8, type="discrete")) +
  theme_minimal(base_size = 12) +
  labs(title = "d", y = "Pathogen damage (%)", caption = "") +
  theme(plot.title = element_text(color="black", size = 14, face="bold")) +
 annotate(geom="text",label=sprintf("italic('p')~'%s'", p2),parse=TRUE,x=-1,y=0.75) +
    guides(color = "none")  +      #To turn off color legend
  guides(fill = "none") + #To turn off fill legend
  guides(shape = "none")
  
  
#pat_pc2

#ggsave(filename="./plots/Aim3_PatPC2_AUG282023.png", plot = pat_pc2, dpi=600, units=c("mm"), width = 180, height = 180, bg = "white")
```

```{r, Figure 3}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false
#| fig.width: 10
#| fig.height: 10
#| fig.cap: "Simple linear regressions of herbivory and pathogen damage on PC1 and PC2 axes from PCAs of leaf traits for ant herbivory and pathogen damage assays. Linear regression of a) percent herbivory damage and PC1 axis (R^2^-adjusted= -0.0024, *p* = 0.447); b) percent herbivory damage and PC2 axis (R^2^-adjusted = 0.079, *p* = <0.001); c) percent pathogen damage and PC1 axis (R^2^-adjusted = 0.064, *p* = <0.001); and d) percent pathogen damage and PC2 axis (R^2^-adjusted  = 0.0016, *p* = 0.207). Colors represent individual species. Circle and triangles represent *E-* and *E+* treatments, respectively."
antpat_arranged <- ggarrange(ant_pc1, ant_pc2, pat_pc1, pat_pc2, nrow = 2, ncol=2, common.legend = TRUE, legend="bottom")

antpat_arranged
# Adding caption
# ant_caption <- expression("*All individual leaves per individual per tree species included ("~ italic("n") ~ "= 210).")
# ant_pca_with_caption <- annotate_figure(antpca_arranged, bottom = text_grob(ant_caption), fig.lab.pos = "bottom.right", fig.lab.size = 6)

#ant_pca_with_caption

#ggsave(filename = "./plots/Aim3_AntPCS_SEPT252023.png", plot = antpat_arranged, dpi=600, units=c("mm"), width = 200, height = 180, bg = "white")
```

## Figure 4

```{r M3REM}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false
#| 
#Optimizers
ctrl <- lmeControl(opt='optim')
ctrl2 <- lmeControl(opt = "nlminb")

#M3REM: Same fixed terms as M2REM
M3REM <- lme(logit_herbivory ~ Thickness  + LMA + E_load, weights = varIdent(form = ~ 1|Species),  random = ~ 1 | Species, method = "REML", control = ctrl, data = masterant2, na.action = na.exclude)

#summary(M3REM)
#AIC(M3REM) #584.711 #1902.209 as of 5/23/23 # 741.7825 as of 30/OCT/2023
```

```{r P3REM}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false
#| 
ctrl <- lmeControl(opt='optim')
ctrl2 <- lmeControl(opt = "nlminb")

P3REM <- lme(logit_pathogenicity ~  LMA + Thickness, weights = varIdent(form = ~ 1|Species),  random = ~ 1 | Species/E_load/Treatment, method = "REML", control = ctrl, data = masterpat2, na.action = na.exclude)
#summary(P3REM)
#AIC(P3REM) #1011.239 as of 30/OCT/2023
```

```{r, Table2, tidy = TRUE}
#| echo: false
#| eval: true
# Regression model table

tab_model(M3REM, P3REM, 
          show.aic = TRUE, 
          show.intercept = TRUE, 
          show.df = FALSE,
          show.ci = 0.95,
          title = "Linear mixed effects models for predicting leaf herbivory and pathogenicity", 
          CSS = list(
            css.depvarhead = 'font-weight:bold;',
            css.centeralign = 'text-align: left;',
            css.firsttablecol = 'font-weight: bold;',
            css.summary = 'color: grey; font-weight: bold;',
            p.style = "numeric_stars"))

```

## Table 1

```{r, table1}
#| echo: false
#| #eval: true
#| tidy: true
#| warning: false
#| tbl-cap: "Summary statistics for the leaf functional traits"
#| 

# Using `gtsummary` and 'flextable' packages
summa_t1 <- tbl_summary(data2 %>%
  group_by(Species, E_load) %>%
  select(c(10:13)) %>%
  rename( Treatment = E_load) %>%
  mutate(Species = case_match(Species, "APEIME" ~ "A. membranacea", 
                            "CHRYCA" ~ "C. cainito", 
                            "CORDAL" ~ "C. alliodora",
                            "DYPTE" ~ "Dypterix .sp",
                            "HEISCO" ~ "H. concinna",
                            "LACPA" ~ "L. panamensis",
                            "THEOCA" ~ "T. cacao")) %>%
  rename('Anthocyanins (ACI)' = Anthocyanins, 
           'Leaf Thickness (LT) (\U00B5m)'=Thickness, 
           'Leaf Punch Strength (LPS) (N mm/-1)'= Toughness, 
           'Leaf Mass per Area (LMA) (mg)' = LMA) %>%
    na.omit(),
  by = Species,
  missing = "no",
  statistic = list(
    all_continuous() ~ "{mean} \u00B1 {sd}", 
    all_categorical() ~ "{n}")) %>%
  modify_header(label = "**Traits/treatment**",
                stat_1 = '**A. membranacea**, n = 83',
                stat_2 = '**C. alliodora**, n = 100',
                stat_3 = '**C. cainito**, n = 150',
                stat_4 = '**Dypterix .sp**, n = 288',
                stat_5 = '**H. concinna**, n = 132',
                stat_6 = '**L. panamensis**, n = 185',
                stat_7 = '**T. cacao**, n = 176',
                text_interpret = "md")
  #modify_footnote(N = "N = count number", abbreviation = TRUE)
  #modify_spanning_header(all_stat_cols(stat_0= T) ~ "**Species**") 
  #modify_caption("**Table 1: Summary statistics for the leaf functional traits**")


#Turning it into a flextable object for further manipulation.
summa_t1 <- summa_t1 %>% 
  as_flex_table() %>%
  # set_caption(
  #   as_paragraph(
  #     as_b(
  #       as_chunk("Table 1: Summary statistics for the leaf functional traits",
  #                props = fp_text_default(font.family = "Tex Gyre Termes", font.size = 12)
  #              )
  #     )
  #   ),
  # align_with_table = F) %>%
  align(j = 1, align = "right", part = "all") %>%
  align(j = 2:8, align = "center", part = "all") %>%
  italic(j = c(2:8), italic = TRUE, part = "header") %>%
  bold(i = 1, bold = TRUE, part = "body") %>%
  #padding(j = 1, padding.right= 40) %>%
  width(j = 1, width = 1.5, unit = "in") %>%
  flextable::font(
    i = NULL,
    j = NULL,
    fontname = "Tex Gyre Termes",
    part = "all",
    cs.family = fontname,
    hansi.family = fontname,
    eastasia.family = fontname) %>%
  autofit()

# Adding some thicker borders to the table
thick <- fp_border(color = "black", style = "solid", width = 2)

summa_t1 <- summa_t1 %>%
  hline_bottom(part = "body", border = thick) %>%
  hline_top(part = "header", border = thick)

#summa_t1

sect_properties <- prop_section(
  page_size = page_size(
    orient = "landscape",
    width = 8.25, height = 11.7
  ),
  type = "continuous",
  page_margins = page_mar()
)

save_as_docx(x = summa_t1, path ="/home/baponte/Boxx/Dissertation/Leaf_Traits_Panama/Data/Aim3_Sequence_analyses/tables/summa_t1.docx", pr_section = sect_properties, align = "center")




```


